plugins/services/src/js/containers/services/ServicesTable.js

```diff
@@ -239,71 +239,78 @@ class ServicesTable extends React.Component {
     const scaleTextID = isGroup
       ? ServiceActionLabels.scale_by
       : ServiceActionLabels[SCALE];
+    const isSDK = isSDKService(service);
 
-    const dropdownItems = [
-      {
-        className: "hidden",
-        id: MORE,
-        html: "",
-        selectedHtml: <Icon id="ellipsis-vertical" size="mini" />
-      },
-      {
-        className: classNames({
-          hidden: !this.hasWebUI(service)
-        }),
+    const actions = [];
+
+    actions.push({
+      className: "hidden",
+      id: MORE,
+      html: "",
+      selectedHtml: <Icon id="ellipsis-vertical" size="mini" />
+    });
+
+    if (this.hasWebUI(service)) {
+      actions.push({
         id: OPEN,
         html: this.props.intl.formatMessage({ id: ServiceActionLabels.open })
-      },
-      {
-        className: classNames({ hidden: isGroup }),
+      });
+    }
+
+    if (!isGroup) {
+      actions.push({
         id: EDIT,
         html: this.props.intl.formatMessage({ id: ServiceActionLabels.edit })
-      },
-      {
-        className: classNames({
-          hidden: (isGroup && instancesCount === 0) || isSingleInstanceApp
-        }),
+      });
+    }
+
+    if ((!isGroup && instancesCount > 0) || !isSingleInstanceApp) {
```

Comment:
This is the only case that we need to use `||` because will not add the `scale` since won't match `!isGroup && instancesCount > 0) && !isSingleInstanceApp` when actually need the `scale`.